fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt

```diff
@@ -309,11 +315,18 @@ class Core(
             // Install the "icons" WebExtension to automatically load icons for every visited website.
             icons.install(engine, this)
 
-            // Install the "ads" WebExtension to get the links in an partner page.
-            adsTelemetry.install(engine, this)
+            CoroutineScope(Dispatchers.Main).launch {
+                val (repo, readJson) = withContext(Dispatchers.IO) {
+                    SerpTelemetryRepository(context.filesDir, COLLECTION_NAME) to
+                        context.assets.readJSONObject("search/search_telemetry_v2.json")
+                }
```

Comment:
Once we move `readJson` to be a parameter to `SerpTelemetryRepository`, this function could look more like
```suggestion
                val readJson = { context.assets.readJSONObject("search/search_telemetry_v2.json") }
                val providerList = withContext(Dispatchers.IO) {
                    SerpTelemetryRepository(
                        rootStorageDirectory = context.filesDir,
                        readJson = readJson,
                        collectionName = COLLECTION_NAME,
                    ).updateProviderList()
                }

                // Install the "ads" WebExtension to get the links in an partner page.
                adsTelemetry.install(engine, this@apply, providerList)

                // Install the "cookies" WebExtension and tracks user interaction with SERPs.
                searchTelemetry.install(engine, this@apply, providerList)
```

This has the added benefit of only running `updateProviderList` once and sharing the result between the two subclasses of `BaseSearchTelemetry`.

Reply:
Definitely! Thanks. Did it for focus as well!